### ISSUE TITLE
java.lang.Throwable: Slow operations are prohibited on EDT. See SlowOperations.assertSlowOperationsAreAllowed javadoc.

### ISSUE THREAD

**Issue by roni-estein** [pestphp/pest-intellij#219]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Plugin version   | 1.2.1-EAP.1
| Pest version     | x.y.z
| IntelliJ version | PS-213.5449.28
| OS               | Mac OS X

### Description
I copied and pasted the full stack trace on this one, I was in iterm doing a mysql restore while this happened.

### Stacktrace
```
java.lang.Throwable: Slow operations are prohibited on EDT. See SlowOperations.assertSlowOperationsAreAllowed javadoc.
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:182)
	at com.intellij.util.SlowOperations.assertSlowOperationsAreAllowed(SlowOperations.java:102)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:784)
	at com.intellij.psi.stubs.StubIndexImpl.getContainingIds(StubIndexImpl.java:514)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:318)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:100)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:88)
	at com.jetbrains.php.PhpIndexImpl.getElements(PhpIndexImpl.java:592)
	at com.jetbrains.php.PhpIndexImpl.lambda$getFunctionsByName$13(PhpIndexImpl.java:540)
	at com.jetbrains.php.PhpIndexImpl.getCached(PhpIndexImpl.java:521)
	at com.jetbrains.php.PhpIndexImpl.getFunctionsByName(PhpIndexImpl.java:539)
	at com.jetbrains.php.PhpIndexImpl.getFunctionsByFQN(PhpIndexImpl.java:548)
	at com.jetbrains.php.PhpIndexImpl.getBySignatureInternal(PhpIndexImpl.java:741)
	at com.jetbrains.php.PhpIndexImpl.getBySignature(PhpIndexImpl.java:680)
	at com.jetbrains.php.PhpIndexImpl.getBySignature(PhpIndexImpl.java:667)
	at com.jetbrains.php.lang.psi.resolve.types.generics.PhpGenericsBaseExtendedWithGenericTypeProvider.decodeExtendedClassesAndSubstitutedTemplates(PhpGenericsBaseExtendedWithGenericTypeProvider.java:160)
	at com.jetbrains.php.lang.psi.resolve.types.generics.PhpGenericsBaseExtendedWithGenericTypeProvider.decodeExtendedClassesAndSubstitutedTemplates(PhpGenericsBaseExtendedWithGenericTypeProvider.java:125)
	at com.jetbrains.php.lang.psi.resolve.types.generics.PhpGenericsBaseExtendedWithGenericTypeProvider.getExtendedClassesToSubstitutedTemplates(PhpGenericsBaseExtendedWithGenericTypeProvider.java:118)
	at com.jetbrains.php.lang.psi.resolve.types.generics.PhpGenericsBaseExtendedWithGenericTypeProvider.complete(PhpGenericsBaseExtendedWithGenericTypeProvider.java:81)
	at com.jetbrains.php.PhpIndexImpl.lambda$doCompleteType$1(PhpIndexImpl.java:165)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:114)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:44)
	at com.jetbrains.php.PhpIndexImpl.doCompleteType(PhpIndexImpl.java:165)
	at com.jetbrains.php.PhpIndexImpl.completeType(PhpIndexImpl.java:120)
	at com.jetbrains.php.PhpIndexImpl.completeType(PhpIndexImpl.java:99)
	at com.jetbrains.php.lang.psi.resolve.types.PhpType.global(PhpType.java:756)
	at com.pestphp.pest.customExpectations.generators.ExpectationGenerator.methodString(ExpectationGenerator.kt:32)
	at com.pestphp.pest.customExpectations.generators.ExpectationGenerator.access$methodString(ExpectationGenerator.kt:11)
	at com.pestphp.pest.customExpectations.generators.ExpectationGenerator$generate$1.invoke(ExpectationGenerator.kt:16)
	at com.pestphp.pest.customExpectations.generators.ExpectationGenerator$generate$1.invoke(ExpectationGenerator.kt:16)
	at kotlin.text.StringsKt__AppendableKt.appendElement(Appendable.kt:85)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinTo(_Collections.kt:3344)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinToString(_Collections.kt:3361)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinToString$default(_Collections.kt:3360)
	at com.pestphp.pest.customExpectations.generators.ExpectationGenerator.generate(ExpectationGenerator.kt:16)
	at com.pestphp.pest.customExpectations.generators.ExpectationGenerator.generateToFile(ExpectationGenerator.kt:42)
	at com.pestphp.pest.customExpectations.ExpectationFileService$generateFile$lambda-3$$inlined$runReadAction$1.compute(actions.kt:60)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:852)
	at com.pestphp.pest.customExpectations.ExpectationFileService.generateFile$lambda-3(ExpectationFileService.kt:87)
	at com.intellij.openapi.project.DumbServiceImpl.lambda$smartInvokeLater$9(DumbServiceImpl.java:530)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:214)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:196)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:805)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:348)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:82)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:131)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:187)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:891)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:760)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:447)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:818)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:446)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:805)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:492)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by olivernybroe** [comment]
Thanks for reporting. 

Errors like these only happens when using EAP or beta versions of intellij. 

I will however still look into it and fix it ✅

**Comment by olivernybroe** [comment]
Should have been fixed in the next version 1.3.0 which will be out soon.
